### PR TITLE
ornl.ascent CI setup

### PR DESCRIPTION
**Pull request by andre-merzky**
Note that this PR is based off @zekemorton 's work in the llnl-ci branch and the diff will only be meaningful once that is merged.  I'll also squash the commits once that has happened.

### CODE REVIEW

File: .gitlab-ci.ornl.yml
Comment (zekemorton) [inline .gitlab-ci.ornl.yml:18]: Love this, will make this change to the llnl pipeline!

File: .gitlab-ci.ornl.yml
Comment (zekemorton) [inline .gitlab-ci.ornl.yml:29]: Do we want the Spack Build to be a test that we upload the the dashboard?

**Comment by zekemorton** [comment]
This looks great! Just a few discussion points so that we can keep the two pipelines consistent with each other! 

It looks like the conda/rockylinux builds are now failing during flux portion of the install process, do we want to remove that combination from the build matrix for now and create a ticket to investigate further?

File: .gitlab-ci.ornl.yml
Comment (andre-merzky) [reply .gitlab-ci.ornl.yml:79]: That makes sense.  I have this disabled for the time being to speed up testing, but will apply your suggestion before merge.  Thanks!

File: .gitlab-ci.ornl.yml
Comment (andre-merzky) [reply .gitlab-ci.ornl.yml:18]: Yeah, not sure how that works for multiple machines - but for now it's quite convenient to have the collected.  Ok to leave it as such until we get a second ORNL machine included?

File: .gitlab-ci.ornl.yml
Comment (andre-merzky) [reply .gitlab-ci.ornl.yml:29]: Given that this is the SDK test suite I would argue that deployment is an important part.  But I am open to removing it also if you think that removes clutter...

**Comment by andre-merzky** [comment]
Hi @zekemorton : I merged the changes from the `llnl-ci` branch and things look ok.  This should best be reviewed though after `llnl-ci` is merged though to reduce the diff footprint.

**Comment by andre-merzky** [comment]
> It looks like the conda/rockylinux builds are now failing during flux portion of the install process, do we want to remove that combination from the build matrix for now and create a ticket to investigate further?

Yes, that might be the best way forward right now.  

GH tests are also complaining about the MongoDB credential I added to the script (and rightly so), so I will have to clean that up also.

**Comment by andre-merzky** [comment]
> > It looks like the conda/rockylinux builds are now failing during flux portion of the install process, do we want to remove that combination from the build matrix for now and create a ticket to investigate further?
> 
> Yes, that might be the best way forward right now.

So what fails is the centos7 / pip / mpich / python-3.7 combination.  I would be inclined to disable python-3.7 for now, as would have the smallest impact on the other tests (I think?) - or do you see a way to disable that specific combination?

**Comment by andre-merzky** [comment]
> > .
> 
> So what fails is the centos7 / pip / mpich / python-3.7 combination. I would be inclined to disable python-3.7 for now, as would have the smallest impact on the other tests (I think?) - or do you see a way to disable that specific combination?

@wilke, @mtitov : never mind that part - the test which previously failed seems to work now  `¯\_(ツ)_/¯`

File: .gitlab-ci.ornl.yml
Comment (mtitov) [inline .gitlab-ci.ornl.yml:1]: general comment for this CI config file - move it into `/.gitlab` directory

File: .gitlab-ci.ornl.yml
Comment (mtitov) [inline .gitlab-ci.ornl.yml:26]: should be in sync with either (1) install and then removed or (2) installed once (in offline) and kept updated. Thus to have an up-to-date copy of Spack
https://github.com/ExaWorks/SDK/blob/04007917a86f77f6f51d0700a6d4507843a9a965/.gitlab/llnl-ci.yml#L8-L17

p.s. also would like to refer to have an option `-c core.sharedRepository=true` and to use `chmod -fR 02770 .` (see [LLNL CI readme](https://github.com/ExaWorks/SDK/blob/master/.gitlab/llnl-readme.md#environment-preparation-for-llnl-machines)) - thus anyone from our group (`exaworks`) can execute these commands. Before that we had permissions issue, since only one user was allowed to make changes

File: ci/tests/parsl/test.sh
Comment (mtitov) [inline ci/tests/parsl/test.sh:17]: something wrong with indentation (?)

File: ci/tests/rp-flux/test.sh
Comment (mtitov) [inline ci/tests/rp-flux/test.sh:8]: should be `flux`, since the file name is `resource_flux`
(btw I've prepared PR #135 to have RP tests updated)

File: .gitlab-ci.ornl.yml
Comment (mtitov) [reply .gitlab-ci.ornl.yml:79]: as a reference to what is discussed in this thread
https://github.com/ExaWorks/SDK/blob/04007917a86f77f6f51d0700a6d4507843a9a965/.gitlab/llnl-ci.yml#L28-L33

File: .gitlab-ci.ornl.yml
Comment (andre-merzky) [reply .gitlab-ci.ornl.yml:1]: Done

File: .gitlab-ci.ornl.yml
Comment (andre-merzky) [reply .gitlab-ci.ornl.yml:26]: Done

File: ci/tests/parsl/test.sh
Comment (andre-merzky) [reply ci/tests/parsl/test.sh:17]: Yeah, someone used tabs - fixed.

File: ci/tests/rp-flux/test.sh
Comment (andre-merzky) [reply ci/tests/rp-flux/test.sh:8]: Fixed.

File: .gitlab/ornl-ci.yml
Comment (mtitov) [inline .gitlab/ornl-ci.yml:17]: `if [ -d $SPACK_WORK_DIR/spack ]; then exit 0; fi`? (if spack-dir exists, then this stage is skipped)

**Review by mtitov — APPROVED** [review]
have one comment only, overall LGTM!

**Comment by andre-merzky** [comment]
I seem to remember that we wanted to disable that rockylinux8 configuration because of the Flux build failure - so merging this anyway...

**Pull request merged**
